jellyfin/jellyfin-androidtv
# add seperate button to adjust subtitle delay (if subtitles are present).

**Changes**


- Add seperate button for vlc player to set subtitle delay.
- Allow default for aformentioned subtitle delay.

**Issues**
fixes #1636 
fixes #1234 
fixes #640

**test**

- Check if button not visible if no subtitles are present.
- Check if subtitles are changed according to setting.
- Check default subtitle.

**further improvement**

It would probably be nicer to share functionality with the audio delay button: but I'm unsure what would be the nicest way to achieve that. My guess would be:

- create a single `delay-popup.xml` instead of seperate `audio-delay-popup.xml` and `subtitle-delay-popup.xml`
- similarly create a single `AdjustDelayAction` instead of two seperate ones
- and 1 `AdjustDelayPopup` which determines if one (or both) settings should be shown & adjusts size accordingly.

Not entirely sure if that would work. And my gut feeling says that there should be a way to keep the two `.xml` files seperate & add a level in between that merges the two. Not entirely sure how though. Input appreciated.
